pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test: TestLogic_enums failed

pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test.TestLogic_enums [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6110895?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6110895?buildTab=artifacts#/) on master @ [32f7d263043cfb47f3ac409a8aaa7346e56b973b](https://github.com/cockroachdb/cockroach/commits/32f7d263043cfb47f3ac409a8aaa7346e56b973b):


```
=== RUN   TestLogic_enums
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/066244445b65e6e62e691d0b977526fa/logTestLogic_enums522635006
    test_log_scope.go:80: use -show-logs to present logs inline
[08:52:19] rng seed: 4052355866114653831

```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_enums.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-18635

```
panic: interface conversion: interface is nil, not nstree.item [recovered]
	panic: interface conversion: interface is nil, not nstree.item

goroutine 641071 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc0052f6000, {0x6cc6858, 0xc0087f2fc0}, {0x5d01060, 0xc009716c90})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1049 +0x3d4
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1({0x6cc6858, 0xc0087f2fc0}, {0xc0052f6000})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:806 +0x5a
panic({0x5d01060, 0xc009716c90})
	GOROOT/src/runtime/panic.go:844 +0x258
github.com/cockroachdb/cockroach/pkg/sql/catalog/nstree.clear(0xc0087971e0)
	github.com/cockroachdb/cockroach/pkg/sql/catalog/nstree/tree.go:65 +0x6a
github.com/cockroachdb/cockroach/pkg/sql/catalog/nstree.byIDMap.clear({0xc0087971e0})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/nstree/by_id_map.go:39 +0x1e
github.com/cockroachdb/cockroach/pkg/sql/catalog/nstree.(*Map).Clear(0xc009fdbbd8)
	github.com/cockroachdb/cockroach/pkg/sql/catalog/nstree/map.go:90 +0x68
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*leasedDescriptors).releaseAll(0xc009fdbbc8, {0x6cc6900, 0xc006c4b170})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/descs/leased_descriptors.go:242 +0x17d
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*Collection).ReleaseLeases(0xc009fdbba0, {0x6cc6900, 0xc006c4b170})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/descs/collection.go:171 +0x3d
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*Collection).ReleaseAll(0xc009fdbba0, {0x6cc6900, 0xc006c4b170})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/descs/collection.go:179 +0x2c
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Cleanup(0xc006bc1c20, {0x6cc6900, 0xc006c4b170})
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:517 +0xb1
github.com/cockroachdb/cockroach/pkg/sql/rowflow.(*rowBasedFlow).Cleanup(0xc00818b890, {0x6cc6900, 0xc006c4b170})
	github.com/cockroachdb/cockroach/pkg/sql/rowflow/row_based_flow.go:440 +0x3c
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run.func6()
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:711 +0x71
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun.func2()
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1570 +0x2e
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc0052f6000, {0x6cc6900, 0xc006c4aa20}, 0xc0052f64f8, 0x3, {0x7fd37abe3808, 0xc0052426c0}, 0x1, 0xc00809cd18)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1561 +0xd2f
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc0052f6000, {0x6cc6900, 0xc006c4aa20}, 0xc0052f64f8, {0x7fd37abe3808, 0xc0052426c0})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1174 +0xf06
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc0052f6000, {0x6cc6900, 0xc006c4a420}, {{0x6ced6d8, 0xc00596e7d0}, {0xc0051bcc9c, 0x20}, 0x0, 0x1}, 0x0, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:699 +0x43a6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc0052f6000, {0x6cc6900, 0xc006c4a210}, {{0x6ced6d8, 0xc00596e7d0}, {0xc0051bcc9c, 0x20}, 0x0, 0x1}, 0x0, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:148 +0xc88
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1({{{0x6ced6d8, 0xc00596e7d0}, {0xc0051bcc9c, 0x20}, 0x0, 0x1}, {0xc0b6f490dc1dd613, 0x637fd08351, 0x0}, {0xc0b6f490dc1dddff, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1893 +0x3af
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc0052f6000)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1897 +0x1088
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc0052f6000, {0x6cc6858, 0xc0087f2fc0}, 0xc0072c5970, 0xc0063a58c0, 0xc006c95dc0)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1819 +0x3d2
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc003684000, {0x6cc6858, 0xc0087f2fc0}, {0xc0052f6000}, 0xc0063a58c0, 0xc006c95dc0)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:808 +0x25e
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1()
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:728 +0x6e5
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:639 +0x2a6
I220816 08:52:19.527720 1 (gostd) testmain.go:650  [-] 1  Test //pkg/sql/logictest/tests/fakedist-vec-off:fakedist-vec-off_test exited with error code 2
```

--- a/pkg/sql/flowinfra/BUILD.bazel
+++ b/pkg/sql/flowinfra/BUILD.bazel
@@ -25,7 +25,6 @@ go_library(
         "//pkg/settings/cluster",
         "//pkg/sql/catalog/colinfo",
         "//pkg/sql/catalog/descpb",
-        "//pkg/sql/catalog/typedesc",
         "//pkg/sql/execinfra",
         "//pkg/sql/execinfra/execopnode",
         "//pkg/sql/execinfra/execreleasable",

--- a/pkg/sql/flowinfra/inbound.go
+++ b/pkg/sql/flowinfra/inbound.go
@@ -192,7 +192,7 @@ func processProducerMessage(
 	draining *bool,
 	msg *execinfrapb.ProducerMessage,
 ) processMessageResult {
-	err := sd.AddMessage(ctx, msg, &flowBase.FlowCtx)
+	err := sd.AddMessage(ctx, msg)
 	if err != nil {
 		return processMessageResult{
 			err: errors.Wrapf(err, "%s",
@@ -225,7 +225,7 @@ func processProducerMessage(
 
 		if log.V(3) && row != nil {
 			if types == nil {
-				types = sd.types()
+				types = sd.Types()
 			}
 			log.Infof(ctx, "inbound stream pushing row %s", row.String(types))
 		}

--- a/pkg/sql/flowinfra/outbox_test.go
+++ b/pkg/sql/flowinfra/outbox_test.go
@@ -152,7 +152,7 @@ func TestOutbox(t *testing.T) {
 			}
 			t.Fatal(err)
 		}
-		err = decoder.AddMessage(context.Background(), msg, &flowCtx)
+		err = decoder.AddMessage(context.Background(), msg)
 		if err != nil {
 			t.Fatal(err)
 		}

--- a/pkg/sql/flowinfra/stream_data_test.go
+++ b/pkg/sql/flowinfra/stream_data_test.go
@@ -81,7 +81,7 @@ func testRowStream(tb testing.TB, rng *rand.Rand, types []*types.T, records []ro
 			msg := se.FormMessage(context.Background())
 			// Make a copy of the data buffer.
 			msg.Data.RawBytes = append([]byte(nil), msg.Data.RawBytes...)
-			err := sd.AddMessage(context.Background(), msg, nil /* flowCtx */)
+			err := sd.AddMessage(context.Background(), msg)
 			if err != nil {
 				tb.Fatal(err)
 			}
@@ -138,7 +138,7 @@ func TestEmptyStreamEncodeDecode(t *testing.T) {
 	var se flowinfra.StreamEncoder
 	var sd flowinfra.StreamDecoder
 	msg := se.FormMessage(context.Background())
-	if err := sd.AddMessage(context.Background(), msg, nil /* flowCtx */); err != nil {
+	if err := sd.AddMessage(context.Background(), msg); err != nil {
 		t.Fatal(err)
 	}
 	if msg.Header == nil {
@@ -220,7 +220,7 @@ func BenchmarkStreamDecoder(b *testing.B) {
 
 			for i := 0; i < b.N; i++ {
 				var sd flowinfra.StreamDecoder
-				if err := sd.AddMessage(ctx, msg, nil /* flowCtx */); err != nil {
+				if err := sd.AddMessage(ctx, msg); err != nil {
 					b.Fatal(err)
 				}
 				for j := 0; j < flowinfra.OutboxBufRows; j++ {

--- a/pkg/sql/flowinfra/stream_decoder.go
+++ b/pkg/sql/flowinfra/stream_decoder.go
@@ -13,8 +13,6 @@ package flowinfra
 import (
 	"context"
 
-	"github.com/cockroachdb/cockroach/pkg/sql/catalog/typedesc"
-	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
 	"github.com/cockroachdb/cockroach/pkg/sql/rowenc"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
@@ -61,11 +59,7 @@ type StreamDecoder struct {
 // msg.Data.Metadata until all the rows in the message are retrieved with GetRow.
 //
 // If an error is returned, no records have been buffered in the StreamDecoder.
-//
-// flowCtx can be nil in tests.
-func (sd *StreamDecoder) AddMessage(
-	ctx context.Context, msg *execinfrapb.ProducerMessage, flowCtx *execinfra.FlowCtx,
-) error {
+func (sd *StreamDecoder) AddMessage(ctx context.Context, msg *execinfrapb.ProducerMessage) error {
 	if msg.Header != nil {
 		if sd.headerReceived {
 			return errors.Errorf("received multiple headers")
@@ -78,16 +72,6 @@ func (sd *StreamDecoder) AddMessage(
 		}
 		sd.typingReceived = true
 		sd.typing = msg.Typing
-		if flowCtx != nil {
-			// Before we can safely use types, we need to make sure they are
-			// hydrated.
-			resolver := flowCtx.NewTypeResolver(flowCtx.Txn)
-			for i := range sd.typing {
-				if err := typedesc.EnsureTypeIsHydrated(ctx, sd.typing[i].Type, &resolver); err != nil {
-					return err
-				}
-			}
-		}
 	}
 
 	if len(msg.Data.RawBytes) > 0 {
@@ -172,9 +156,9 @@ func (sd *StreamDecoder) GetRow(
 	return rowBuf, nil, nil
 }
 
-// types returns the types of the columns; can only be used after we received at
+// Types returns the types of the columns; can only be used after we received at
 // least one row.
-func (sd *StreamDecoder) types() []*types.T {
+func (sd *StreamDecoder) Types() []*types.T {
 	types := make([]*types.T, len(sd.typing))
 	for i := range types {
 		types[i] = sd.typing[i].Type

